gprofiler/profilers/python.py

```diff
@@ -215,6 +224,10 @@ def _profile_process(self, process: Process, duration: int, spawned: bool) -> Pr
         )
         appid = application_identifiers.get_python_app_id(process)
         app_metadata = self._metadata.get_metadata(process)
+        if self._container_names_client:
+            container_name = self._container_names_client.get_container_name(process.pid)
+        else:
+            container_name = ""
```

Comment:
This logic repeats itself - let's encapsulate to a helper. This helper can be a part of the refactor I suggested in https://github.com/Granulate/gprofiler/pull/667/files#r1091286777 - the `ProfilerState`, or `GProfiler`, can expose a method `get_container_name(pid: int)` which gets you the name or `""`.